File: mlua-gen/tests/tuple_nested.lua
local p = Pair(Inner { x = 1 }, { 10, 20, 30 })

assert(p[1].x == 1)
p[1].x = 99
assert(p[1].x == 99)

assert(p[2][1] == 10)
assert(p[2][2] == 20)
p[2][1] = 11
assert(p[2][1] == 11)

File: mlua-gen/tests/tuple_nested.rs
//! Tuple struct: nested mlua_gen field and indexable collection field.

use mlua_gen::{LuaBuilder, mlua_gen};

#[test]
pub fn test() -> mlua::Result<()> {
    #[derive(Debug, Clone)]
    #[mlua_gen]
    struct Inner {
        pub x: u32,
    }

    #[mlua_gen]
    struct Pair(pub Inner, pub Vec<u8>);

    let lua = mlua::Lua::new();
    Inner::to_globals(&lua)?;
    Pair::to_globals(&lua)?;

    lua.load(include_str!("./tuple_nested.lua")).exec()?;
    Ok(())
}
